api/api.proto

```diff
@@ -73,14 +73,42 @@ AppletUpdate
 A `AppletUpdate` represents an OTA sequence applet slice.
 
 The `TotalChunks` value indicates the total number of chunks for the update,
-`Seq` is the transmitted `Data` chunk number (starting with 1 as the value for
-the first chunk).
+`Seq` is the transmitted AppletUpdate chunk number:
+ - `0` indicates that the struct contains verification data in `Header`.
+ - `1` onwards identifies the first, second, ... chuck of firmware image data.
 
 */
 message AppletUpdate {
 	uint32 Total = 1;
 	uint32 Seq   = 2;
-	bytes  Data  = 3;
+	oneof Payload {
+		bytes  Data  = 3;
+		AppletUpdateHeader Header = 4;
+	};
+}
+
+
+/*
+
+AppletUpdateHeader
+
+An `AppletUpdateHeader` contains verification material for an applet update.
+It's sent in the zero-th `AppletUpdate` message during an OTA.
+
+
+*/
+
+message AppletUpdateHeader {
+	// Signature holds the signature over the applet.
+	bytes  Signature  = 1;
```

Comment:
Whose signature is this? Isn't this already in the Manifest?

Reply:
This is just the existing signature-over-applet made a bit more explicit (it was previously smuggled in via `Data` if `Seq == 0`).

I'd expect it's the manifest itself which will (eventually) be signed - I guess we could use a `note` format for that and have it "attached" if we wanted), or this `Signature` field could be repurposed for that.